src/routes/users/get_default_avatar.rs

```diff
@@ -1,43 +1,76 @@
-            NamedFile::open(Path::new("assets/user_red.png")).await.ok().map(|n| CachedFile(n))
-        }
-        '8' | '9' | 'A' | 'C' | 'B' | 'D' | 'E' | 'F' => {
-            NamedFile::open(Path::new("assets/user_green.png"))
-                .await
-                .ok().map(|n| CachedFile(n))
-        }
-        'G' | 'H' | 'J' | 'K' | 'M' | 'N' | 'P' | 'Q' => {
-            NamedFile::open(Path::new("assets/user_blue.png"))
-                .await
-                .ok().map(|n| CachedFile(n))
-        }
-        'R' | 'S' | 'T' | 'V' | 'W' | 'X' | 'Y' | 'Z' => {
-            NamedFile::open(Path::new("assets/user_yellow.png"))
-                .await
-                .ok().map(|n| CachedFile(n))
-        }
-        _ => unreachable!(),
+impl rocket_okapi::response::OpenApiResponderInner for CachedFile {
+    fn responses(
+        _gen: &mut rocket_okapi::gen::OpenApiGenerator,
+    ) -> std::result::Result<openapi3::Responses, rocket_okapi::OpenApiError> {
+        let mut responses = schemars::Map::new();
+        let mut content = schemars::Map::new();
+
+        content.insert(
+            "image/png".to_owned(),
+            MediaType {
+                schema: Some(SchemaObject {
+                    instance_type: Some(SingleOrVec::Single(Box::new(InstanceType::String))),
+                    format: Some("binary".to_owned()),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            },
+        );
+
+        responses.insert(
+            "200".to_string(),
+            RefOr::Object(openapi3::Response {
+                description: "Default Avatar Picture".to_string(),
+                content,
+                ..Default::default()
+            }),
+        );
+
+        Ok(openapi3::Responses {
+            responses,
+            ..Default::default()
+        })
     }
 }
+
+/// # Fetch Default Avatar
+///
+/// This returns a default avatar based on the given id.
+#[openapi(tag = "User Information")]
+#[get("/<target>/default_avatar")]
+pub async fn req(target: String) -> CachedFile {
+    CachedFile((
+        ContentType::PNG,
+        match target.chars().last().unwrap() {
+            // 0123456789ABCDEFGHJKMNPQRSTVWXYZ
+            '0' | '1' | '2' | '3' | 'S' | 'Z' => {
+                include_bytes!("../../../assets/user/2.png").to_vec()
+            }
+            '4' | '5' | '6' | '7' | 'T' => include_bytes!("../../../assets/user/3.png").to_vec(),
+            '8' | '9' | 'A' | 'B' => include_bytes!("../../../assets/user/4.png").to_vec(),
+            'C' | 'D' | 'E' | 'F' | 'V' => include_bytes!("../../../assets/user/5.png").to_vec(),
+            'G' | 'H' | 'J' | 'K' | 'W' => include_bytes!("../../../assets/user/6.png").to_vec(),
+            'M' | 'N' | 'P' | 'Q' | 'X' => include_bytes!("../../../assets/user/7.png").to_vec(),
+            /*'0' | '1' | '2' | '3' | 'R' | 'Y'*/
+            _ => include_bytes!("../../../assets/user/1.png").to_vec(),
```

Comment:
This could be changed to something like `include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/assets/user/1.png"))` to make it more clear where assets is located.

An `include_asset` macro might help with the verbosity that.

This is more of a personal style preference than anything.

Reply:
Added an `asset!()` macro for resolving the path.